pkg/infra/client.go

```diff
@@ -45,44 +47,45 @@ func CreateGRPCClient(node Node) (*comm.GRPCClient, error) {
 	return grpcClient, nil
 }
 
-func CreateEndorserClient(node Node) (peer.EndorserClient, error) {
-	gRPCClient, err := CreateGRPCClient(node)
+func CreateEndorserClient(node Node, logger *log.Logger) (peer.EndorserClient, error) {
+	conn, err := DailConnection(node, logger)
 	if err != nil {
 		return nil, err
 	}
-
-	conn, err := gRPCClient.NewConnection(node.Addr, func(tlsConfig *tls.Config) { tlsConfig.InsecureSkipVerify = true })
-	if err != nil {
-		return nil, errors.Wrapf(err, "error connecting to %s", node.Addr)
-	}
-
 	return peer.NewEndorserClient(conn), nil
 }
 
-func CreateBroadcastClient(node Node) (orderer.AtomicBroadcast_BroadcastClient, error) {
-	gRPCClient, err := CreateGRPCClient(node)
+func CreateBroadcastClient(node Node, logger *log.Logger) (orderer.AtomicBroadcast_BroadcastClient, error) {
+	conn, err := DailConnection(node, logger)
 	if err != nil {
 		return nil, err
 	}
+	return orderer.NewAtomicBroadcastClient(conn).Broadcast(context.Background())
+}
 
-	conn, err := gRPCClient.NewConnection(node.Addr, func(tlsConfig *tls.Config) { tlsConfig.InsecureSkipVerify = true })
+func CreateDeliverFilteredClient(node Node, logger *log.Logger) (peer.Deliver_DeliverFilteredClient, error) {
+	conn, err := DailConnection(node, logger)
 	if err != nil {
-		return nil, errors.Wrapf(err, "error connecting to %s", node.Addr)
+		return nil, err
 	}
-
-	return orderer.NewAtomicBroadcastClient(conn).Broadcast(context.Background())
+	return peer.NewDeliverClient(conn).DeliverFiltered(context.Background())
 }
 
-func CreateDeliverFilteredClient(node Node) (peer.Deliver_DeliverFilteredClient, error) {
+// TODO: use a global get logger function instead inject a logger
```

Comment:
why? isn't logger injection better than global getter? so you could add some customizable tags, etc